Otus-DevOps-2018-02/ifqthenp_microservices
# HW 21. Logging and distributed tracing (logging-1)

# Completed Homework №21


## The following tasks have been completed

- :large_blue_diamond: Configured and launched centralised EFK logging system:
  - ElasticSearch (searching engine for log data)
  - Fluentd (logs aggregator)
  - Kibana (visualisation of logged data)
- :large_blue_diamond: Configured `post` and `ui` services to send logs to Fluentd service using Docker `fluentd` driver
- :large_blue_diamond: Launched Zipkin distributed tracing service and analysed the traces of services using Zipkin Web UI
- :large_orange_diamond: Configured Fluentd service to parse unstructured logs received from `ui` service with `grok` parser

## How to get project running

- Create new GCE instance using `docker-machine` command:

```shell
docker-machine create --driver google \
  --google-machine-image https://www.googleapis.com/compute/v1/projects/ubuntu-os-cloud/global/images/family/ubuntu-1604-lts \
  --google-machine-type n1-standard-1 \
  --google-disk-size 20 \
  --google-open-port 5601/tcp \
  --google-open-port 9292/tcp \
  --google-open-port 9411/tcp \
  docker-host
```

- Run `make build` from project root
- Change into `docker` directory from project root
- Run `docker-compose up -d`
- Run `docker-compose -f docker-compose-logging.yml up -d`

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Run `docker-machine ip docker-host` to get the public IP of your GCE instance
- Type in browser URL bar:
  - `http://<PUBLIC_IP>:9292` for Reddit application
  - `http://<PUBLIC_IP>:5601` for Kibana service
  - `http://<PUBLIC_IP>:9411` for Zipkin service

## PR checklist
